main.c

```diff
@@ -747,6 +748,8 @@ int main(int argc, char **argv) {
 	cuid = zmalloc(MINIBUF_SIZE);
 	cauth = zmalloc(MINIBUF_SIZE);
 
+	init_auth(g_creds);
```

Comment:
Will this also work if the credentials change? Now when my Kerberos ticket expires, I can run `kinit` enter my password and everything works again without having to restart cntlm.

Reply:
`g_creds` is a pointer to a global structure of type `auth_s`. In the original code this structure is allocated on the heap (malloc) at the beginning of main() and freed at the end of main() (before exiting the program). User info can change inside this variable but the variable itself never changes.

Here both SonarCloud and Scan-build complain that that there are no guarantees that the pointer is not NULL. In practice it is always not NULL, but if malloc() failed inside the `new_auth()` function, it would return a NULL pointer and go on with it.

Being `g_creds` a pointer to a global variable, there is no need for this variable to be allocated on the heap, it can be statically instanced in the static data of the program. The overall behaviour of the program doesn't change, but now we have the guarantee that the global pointer always points to a valid structure.

Reply:
Alright, I indeed didn't understand the impact. Thanks for clarifying!